tpot/operators/classifiers/base.py

```diff
@@ -57,8 +56,22 @@ def _train_and_predict(self, input_matrix, operator_args):
         clf.fit(self.training_features, self.training_classes)
 
         all_features = np.delete(input_matrix, non_feature_columns, axis=1)
-        input_matrix[:, GUESS_COL] = clf.predict(all_features)
+        if self.clf_eval_func == 'decision_function':
+            if 'decision_function' in dir(clf):
+                input_matrix = np.insert(input_matrix, GUESS_COL, clf.decision_function(all_features).T, axis=1)
```

Comment:
You all helped me realize that I was totally overthinking it/ignoring the bigger picture. I think a solution to make this work is to still insert the classifier's `predict()` outputs in GUESS_COL, but to tack on anything regarding `predict_proba()` or `decision_function()` to the end of the matrix, and to use that slice at the end in the scoring functions as necessary. I've updated accordingly.